### PR TITLE
Admin edit static pages

### PR DESCRIPTION
**Pull request by sandlerben**
r @nancywong @kyle-rosenbluth @aharelick

### CODE REVIEW

File: app/models/__init__.py [hunk 1]
```diff
@@ -6,3 +6,4 @@
 from agency import *  # noqa
 from user import *  # noqa
 from incident_report import *  # noqa
+from miscellaneous import *  # noqa
```
Comment (rileynwong) [inline app/models/__init__.py:9]: why `miscellaneous` and not `editableHTML`? what other misc classes would potentially be here in `miscellaneous.py`?

File: app/admin/views.py [hunk 2]
```diff
@@ -231,3 +231,24 @@ def change_agency_public_status(agency_id):
     form.process()
     return render_template('admin/manage_agency.html', agency=agency,
                            form=form)
+
+
+@admin.route('/_update_editor_contents', methods=['POST'])
```
Comment (aharelick) [inline app/admin/views.py:236]: I'm assuming this is proper practice to start the route with an underscore?

File: app/admin/views.py [hunk 3]
```diff
@@ -231,3 +231,24 @@ def change_agency_public_status(agency_id):
...
+@login_required
+@admin_required
+def update_editor_contents():
+    """Update the contents of an editor."""
+
+    edit_data = request.form.get('edit_data')
+    editor_name = request.form.get('editor_name')
+
+    editor_contents = EditableHTML.query.filter_by(
```
Comment (aharelick) [inline app/admin/views.py]: why aren't you using the `get_editable_html()` helper method for this query?

File: app/templates/macros/page_macros.html [hunk 4]
```diff
@@ -0,0 +1,49 @@
+{% macro render_inline_editor(editable_html_obj, current_user) %}
+    <div id="editor-{{ editable_html_obj.editor_name }}" contenteditable="false">
+        {{ editable_html_obj.value|safe }}
+    </div>
+
+    {% if current_user.is_admin() %}
+       <button class="ui button start-edit">
+        Edit
```
Comment (aharelick) [inline app/templates/macros/page_macros.html]: Nit: should this be indented? you could probably put the whole thing on one line...
`<button class="ui button start-edit">Edit</button>`

File: app/templates/macros/page_macros.html [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+       </button>
+       <button class="ui primary button end-edit">
+        Save
```
Comment (aharelick) [inline app/templates/macros/page_macros.html]: Same comment as above about indentation?

File: app/templates/macros/page_macros.html [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+       </button>
+    {% endif %}
+
+    <script>
+    var editorIDName = "editor-{{ editable_html_obj.editor_name }}";
```
Comment (aharelick) [inline app/templates/macros/page_macros.html]: might want to indent everything inside the script tag by one 'tab'

File: app/templates/macros/page_macros.html [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        });
+
+        $(".end-edit").click(function() {
+            if ( CKEDITOR.instances[editorIDName] ) {
+                var json_data = {
+                    csrf_token: "{{ csrf_token() }}",
+                    editor_name: "{{ editable_html_obj.editor_name }}",
+                    edit_data: CKEDITOR.instances[editorIDName].getData(),
+                };
+                $.post("{{ url_for('admin.update_editor_contents') }}", json_data);
```
Comment (aharelick) [inline app/templates/macros/page_macros.html]: you might want to post some kind of flash if the update was successful and display something else if anything other than 200 was returned?

**Comment by aharelick** [comment]
@sandlerben, this looks really awesome and clean. I just had a couple of formatting comments and then one question about giving the user feedback after the POST request responds. Besides that looks pretty cool.

File: app/admin/views.py (hunk 2 above)
Comment (sandlerben) [inline app/admin/views.py:236]: I learned the style from @weronica but I think it's a good practice so it's obvious which routes are not meant to be accessed by the user

Comment (aharelick) [inline app/admin/views.py:236]: :+1:

File: app/models/__init__.py (hunk 1 above)
Comment (sandlerben) [inline app/models/__init__.py:9]: @nancywong Not sure but seemed a bit odd to give this one model it's own python file. What do you think?

**Comment by sandlerben** [comment]
@nancywong @kyle-rosenbluth @aharelick did another round of edits, let me know if it looks good now

File: app/models/__init__.py (hunk 1 above)
Comment (rileynwong) [inline app/models/__init__.py:9]: no strong feelings so otherwise lgtm

**Comment by rileynwong** [comment]
:ship:

**Pull request merged**